### ISSUE TITLE
Label and taints for node pools

### ISSUE THREAD

**Issue by PhilippeVienne** [ovh/public-cloud-roadmap#87]
As a Managed Kubernetes User, I want to be able to set label and taints applied to all nodes inside a given Node Pool.

This will help to have auto-scaled tainted and labeled nodes inside clusters.

**Comment by mhurtrel** [comment]
Hi @PhilippeVienne 
I confirm we will work on this feature. Here is the current implementation we have in mind  : 
- Manage and enforce taints based on IaaS related information (the flavor for example)
- allow you to define labales ans automatically spread them on all current and future nodes in the pool. Can you confirm that would match you needs ?

**Comment by PhilippeVienne** [comment]
Hi @mhurtrel

taints must also be spread in a nodepool, because you can use it to split your cluster workload and avoid certain nodes to take some basic cluster workload

taints are not only based on IaaS information because you must also decide the behaviour to perform on the scheduling

**Comment by mhurtrel** [comment]
Agreed :)

**Comment by mhurtrel** [comment]
s a woraound before we release that improvment, my colleague Xavier suggest you hav a look on https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/ , and  set up an AdmissionController to mutate news nodes automatically on certain conditions

**Comment by mhurtrel** [comment]
We plan to offer this feature in October

**Comment by winstef** [comment]
@mhurtrel just checking if it's still planned for an October release or if there's an ETA?
I'm really looking forward to try it :)

**Comment by mhurtrel** [comment]
Hello @winstef . The feature as been mostly developped, an this is now a question of weeks for production.
It should be put in production within a couple of weeks

**Comment by mhurtrel** [comment]
The feature is now in production ! You can now add/edit labels to nodepool CRD  that will be automatically propagated to each nodes of that pool.
My colleague [@scraly](https://github.com/scraly) will soon propose a tutorial showing an example on how to leverage the feature

**Comment by rverchere** [comment]
> The feature is now in production ! You can now add/edit labels to nodepool CRD that will be automatically propagated to each nodes of that pool. My colleague [@scraly](https://github.com/scraly) will soon propose a tutorial showing an example on how to leverage the feature

Hi, I do not see this option in nodepool creation / edition. Any idea?

**Comment by matmicro** [comment]
Hi,

Could you please share the tutorial link if available ?

Thanks

**Comment by rverchere** [comment]
> Hi,
> 
> Could you please share the tutorial link if available ?
> 
> Thanks

The only documentation I see is https://docs.ovh.com/au/en/kubernetes/taint-drain-node-pools/#taint-a-node, but this is vanilla kubectl commands, not through the manager/api.

**Comment by scraly** [comment]
Hi @rverchere ,
you can find a documentation that show [how to add labels, annotation, taints on node pool through Terraform and the API](https://docs.ovh.com/gb/en/kubernetes/automatically-label-taint-node-pool/) :)

**Comment by fidoriel** [comment]
Hey,
I was wondering why this cannot be configured through UI, This would be great and make things easier. Also I noticed, I cannot create taints and labels through the Custom Resource Definition, which would also be helpful (I did not get an error, but nothing happened), despite the schema allowing it and the crd not throwing an error.
If the node type can be configured, why not the taints and labels. This would be very helpful to us.
